src/Extensions/Diagnostics.HealthChecks/Internal/RestClientWrapper.cs

```diff
@@ -0,0 +1,38 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+using Microsoft.ServiceFabric.Client;
+using Microsoft.ServiceFabric.Client.Http;
+
+namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
+{
+	internal class RestClientWrapper
+	{
+		private ServiceFabricHttpClient? m_client;
+
+		private readonly Uri m_clusterEndpoint;
+
+		public RestClientWrapper(Uri clusterEndpoint)
+		{
+			// m_client is initialized during first publish async in order to avoid possible deadlocks
+			m_clusterEndpoint = clusterEndpoint;
+		}
+
+		public async Task<ServiceFabricHttpClient> GetAsync()
+		{
+			if (m_client == null)
+			{
+				ConfiguredTaskAwaitable < IServiceFabricClient > result = new ServiceFabricClientBuilder()
+					.UseEndpoints(m_clusterEndpoint)
+					.BuildAsync().ConfigureAwait(false);
+
+				m_client = (ServiceFabricHttpClient?)await result;
+			}
+
+			return m_client!;
+		}
```

Comment:
I would suggest something like this instead to avoid dangerous casting (line 32) and dismissing nullable warning (line 35), but I have not validated it with the compiler, please let me know if it won't compile and we could search for a solution.
```suggestion
		private IServiceFabricClient? m_client;

		private readonly Uri m_clusterEndpoint;

		public RestClientWrapper(Uri clusterEndpoint)
		{
			// m_client is initialized during first publish async in order to avoid possible deadlocks
			m_clusterEndpoint = clusterEndpoint;
		}

		public async Task<ServiceFabricHttpClient> GetAsync()
		{
			if (m_client == null)
			{
				m_client = await new ServiceFabricClientBuilder()
					.UseEndpoints(m_clusterEndpoint)
					.BuildAsync()
					.ConfigureAwait(false);
			}

			return m_client;
		}
```